### PR TITLE
Various micro-optimizations in the URL code

### PR DESCRIPTION
**Pull request by MaxKellermann**

### CODE REVIEW

**Comment by fritsch** [comment]
jenkins build this please

**Comment by MrMC** [comment]
you could make the in-lines all on one line to match code convention used in other classes. Otherwise, the code convention is "{" and "}" on separate lines :)

File: xbmc/music/MusicDatabase.cpp [hunk 1]
```diff
@@ -4043,9 +4033,9 @@ bool CMusicDatabase::GetAlbumsByWhere(const std::string &baseDir, const Filter &
     // get data from returned rows
     items.Reserve(results.size());
     const dbiplus::query_data &data = m_pDS->get_result_set().records;
-    for (DatabaseResults::const_iterator it = results.begin(); it != results.end(); ++it)
+    for (const auto &i : results)
     {
-      unsigned int targetRow = (unsigned int)it->at(FieldRow).asInteger();
+      unsigned int targetRow = (unsigned int)i.at(FieldRow).asInteger();
```
Comment (Paxxi) [inline xbmc/music/MusicDatabase.cpp]: Would you mind also updating lines you've touched to use c++ style casts? Can be painful with several commits so this is me asking nicely, not demanding :)

**Comment by Paxxi** [comment]
Nice work @MaxKellerman, always nice getting some modernization done. The inlining of methods will probably be a noop, compiler should figure this out better than us. It doesn't hurt either, but if it's to be merged it needs to follow coding style as @davilla mentioned

**Comment by MaxKellermann** [comment]
On 2016/07/04 21:46, Pär Björklund <notifications@github.com> wrote:
> Nice work @MaxKellerman, always nice getting some modernization
> done. The inlining of methods will probably be a noop, compiler
> should figure this out better than us.

No, that's impossible.  The compiler can't optimize across compilation
units.

(That is, unless you use LTO, which is a workaround for the
"compilation unit" thing.)

**Comment by MaxKellermann** [comment]
Second round:
- fixed the `TestBasicEnvironment` build failure, which was not caused, only triggered by my include cleanups;
- dropped the `std::to_string()` patch (for now), because `std::to_string()` is broken on the old NDK your Jenkins server is using

**Comment by MaxKellermann** [comment]
Third round: another attempt for the `TestBasicEnvironment` build failure, which turns out to be a gcc 4.8 bug (I don't have such an old buggy version, but your build server does). Workaround: add an include dependency which should not be necessary ...

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Comment by MartijnKaijser** [comment]
Any objections?

**Comment by MartijnKaijser** [comment]
@MaxKellermann here's a run with test enabled http://jenkins.kodi.tv/job/TestMulti-All/668/

**Comment by Paxxi** [comment]
👍

**Pull request merged**